packages/dashboards/addon/components/dashboard-dimension-selector.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright 2019, Yahoo Holdings Inc.
+ * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
+ *
+ *  * Usage:
+ *   {{dashboard-dimension-selector
+ *      dashboard=dashboardModel
+ *      onChange=(action changeAction)
+ *   }}
+ */
+import Component from '@ember/component';
+import { computed, get } from '@ember/object';
+import layout from '../templates/components/dashboard-dimension-selector';
+
+export default Component.extend({
+  layout,
+  /**
+   * @property {Promise} -- creates powerselect options of all dimensions that can be pick based on widgets on the dashboard
+   */
+  groupedDimensions: computed('dashboard', 'dashboard.widgets', function() {
+    const widgetPromises = get(this, 'dashboard.widgets');
+    /*
+     * get a list of dimensions per table/timeGrain involved
+     * do this so each table/timegrain combination is unique and we don't have to flatten more than we have to.
+     * shape will be: {table+timeGrain: [{name, longName, catetory}, ...], ...}
+     */
+    return widgetPromises
```

Comment:
Can we break this up into multiple cps or methods?

Reply:
done